packages/core/src/integrations/extraerrordata.ts

```diff
@@ -0,0 +1,86 @@
+import { addGlobalEventProcessor, getCurrentHub } from '@sentry/hub';
+import { Integration, SentryEvent, SentryEventHint } from '@sentry/types';
+import { isError } from '@sentry/utils/is';
+
+/**
+ * Just an Error object with arbitrary attributes attached to it.
+ */
+interface ExtendedError extends Error {
+  [key: string]: any;
+}
+
+/** Patch toString calls to return proper name for wrapped functions */
+export class ExtraErrorData implements Integration {
+  /**
+   * @inheritDoc
+   */
+  public name: string = ExtraErrorData.id;
+
+  /**
+   * @inheritDoc
+   */
+  public static id: string = 'ExtraErrorData';
+
+  /**
+   * @inheritDoc
+   */
+  public setupOnce(): void {
+    addGlobalEventProcessor(async (event: SentryEvent, hint?: SentryEventHint) => {
+      const self = getCurrentHub().getIntegration(ExtraErrorData);
+
+      if (!self || !hint || !hint.originalException) {
+        return event;
+      }
+
+      return this.enhanceEventWithErrorData(event, hint.originalException);
+    });
+  }
+
+  /**
+   * Attaches extracted information from the Error object to extra field in the SentryEvent
+   */
+  public enhanceEventWithErrorData(event: SentryEvent, error: Error): SentryEvent {
+    const errorData = this.extractErrorData(error);
+
+    if (errorData) {
+      return {
+        ...event,
+        extra: {
+          ...event.extra,
+          ...errorData,
+        },
+      };
+    } else {
+      return event;
+    }
+  }
+
+  /**
+   * Extract extra information from the Error object
+   */
+  private extractErrorData(error: ExtendedError): { [key: string]: any } | null {
+    // We are trying to enhance already existing event, so no harm done if it won't succeed
+    try {
+      const name = error.name || error.constructor.name;
+      const errorKeys = Object.keys(error).filter(key => !(key in ['name', 'message', 'stack']));
```

Comment:
I know this is copy/paste but the `in` operator usually doesn't work with arrays?

Reply:
In fact, I believe this isn't really needed because `Object.keys` doesn't return `name`, `message`, and `stack`.
At the same time Object.keys doesn't capture properties such as `lineNumber` in firefox's `SyntaxError`.

`Object.getOwnPropertyDescriptors` might be better suited for fetching properties like that.

Reply:
And you are totally right about `in` usage here, however, we should keep the filter in place, just fixed one. The issue here is that different browsers return different keys (how lovely, right? :p)

```
Chrome:
Object.keys: ["something"]
Object.getOwnPropertyDescriptors: {stack: {…}, message: {…}, something: {…}}

Firefox:
Object.keys: ["something"]
Object.getOwnPropertyDescriptors: { fileName: {…}, lineNumber: {…}, columnNumber: {…}, message: {…}, something: {…} }

Safari:
Object.keys: ["something", "line", "column"]
Object.getOwnPropertyDescriptors: {message: Object, something: Object, line: Object, column: Object, stack: Object}
```

```js
var foo = new SyntaxError('asd');
foo.something = 42;
var nativeKeys = ['name', 'message', 'stack', 'line', 'column', 'fileName', 'lineNumber', 'columnNumber'];
Object.keys(foo).filter(key => nativeKeys.indexOf(key) === -1);
```

This seems to be the most unified way now.